aether-kernel/aether/kernel/api/models.py

```diff
@@ -279,6 +315,8 @@ class Entity(ExportModelOperationsMixin('kernel_entity'), models.Model):
 
     projectschema = models.ForeignKey(to=ProjectSchema, on_delete=models.SET_NULL, null=True)
     submission = models.ForeignKey(to=Submission, on_delete=models.SET_NULL, blank=True, null=True)
+    mapping = models.ForeignKey(to=Mapping, on_delete=models.SET_NULL, null=True)
+    mapping_revision = models.TextField(blank=False, null=False)
 
     # redundant but speed up queries
     project = models.ForeignKey(to=Project, on_delete=models.SET_NULL, blank=True, null=True)
```

Comment:
This is not in this PR scope but I think that a better definition could be with the CASCADE option
`project = models.ForeignKey(to=Project, on_delete=models.CASCADE, blank=True, null=True)`

Otherwise we have orphan entities if the project is deleted, and with the integration tests means, that we are not properly cleaning the database :scream:

Reply:
From what i've gathered so far. I assume entities are not to be automatically deleted at any point. Am i missing something?

Reply:
No, it's part of my eternal _how to create even more confusion_ task... just wondering if it makes sense to have orphan entities, linked to nothing at all.